### ISSUE TITLE
[MS] Define list of directories that need to be excluded from the automatic cleanup

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#10718]
**Impact of the new feature**
MSUnmerged

**Is your feature request related to a problem? Please describe.**
Which the service flexibility implemented in https://github.com/dmwm/WMCore/pull/10706
we should now identify and discuss which directories should be excluded from the automated MSUnmerged.

We need to identify both generic and RSE-specific paths. For instance, CERN is a special case because it also stores WMAgent job logs.

It's possible that SAM tests use a specific unmerged path, so we need to remove those from the deletion. More info at: https://ggus.eu/index.php?mode=ticket_info&ticket_id=153024

As discussed with Todor in the last days, this is the initial list of paths at CERN that we decided to skip:
```
data.dirFilterExcl = ['/store/unmerged/logs/prod/',
                     '/store/unmerged/data/',
                     '/store/unmerged/hidata/',
                     '/store/unmerged/express/']
```
but it could be that we could go one level lower in a few of those cases and actually delete very old log files.

**Describe the solution you'd like**
Identify which directory paths need to be excluded from the MSUnmerged automatic deletion, and provide the necessary configuration changes in both deployment and services_config repositories.

Once those have been properly identified and fixed in the configuration, we might then want to re-enable some of the RSEs that are currently being skipped in the service.

**Describe alternatives you've considered**
If we have something generic, we might also consider excluding such paths from the very beginning, with the unmerged scanner executed by Rucio ConMon.

**Additional context**
None

**Comment by amaltaro** [comment]
On Aug 4, around 5pm CERN time, we have deployed the following service configuration in production:
https://gitlab.cern.ch/cmsweb-k8s/services_config/-/merge_requests/106

as we can see, these 3 directories are defined to be skipped from every single RSE:
```
data.dirFilterExcl = ['/store/unmerged/data/logs/prod/2021',
                      '/store/unmerged/logs/prod/2021',
                      '/store/unmerged/express/logs/prod/2021']
```

this is not final though and we should keep discussing/investigating what needs to be removed or not from the automatic deletion.

**Comment by drkovalskyi** [comment]
Here is a quick summary of the discussion that we had at the Tier0 meeting:
- We can temporarily enable T2_CH_CERN cleaning with the logs in the exception list
  - We need to review the log placement policy - having absolute path has many drawbacks 
- Long term we need to get the active file information from Tier0 instances of WMStats (both production and testbed to cover replays).

**Comment by nsmith-** [comment]
A few paths that arise suspicion (but I don't know if they are needed or not) are
```
/store/unmerged/SAM/
/store/unmerged/tier0_harvest/
```

**Comment by todor-ivanov** [comment]
Thanks @nsmith- , 

For: 
```
/store/unmerged/tier0_harvest/
```
Dima @drkovalskyi did confirm we need to do no additional protection to T0 unmerged area in T2_CH_CERN. At least not until T0 starts using T1 sites too. For which we are working to start reading the T0 wmstatsServer too. But this thing deserves its own issue. 

For:
```
/store/unmerged/SAM/
```
I have no idea, whether we  should protect those or not.

**Comment by amaltaro** [comment]
Just a note: if I'm looking at the right configuration, the Rucio unmerged scanner has this configuration:
https://github.com/dmwm/CMSKubernetes/blob/master/helm/rucio-consistency/templates/unmerged-configmap.yaml#L10-L11

Should we update that configuration to the upper SAM directory? As you suggested, Nick.

**Comment by amaltaro** [comment]
@todor-ivanov is this issue something that we should consider as done now? If so, can you please close it? Thanks

**Comment by todor-ivanov** [comment]
Hi @amaltaro. In general we can close it. I was  keeping it opened only to check out the list yet again once the issues with the `gfal2` memory overflow are resolved. Anyway, we can close it now and reopen later in case further discussion is needed, but I doubt it.

**Comment by amaltaro** [comment]
Thanks, Todor. Perhaps we should keep this open waiting for:
* wait to have unmerged cleanup 100% functional in all RSEs
* once that happens, we update the list of directories to (to avoid issues in 2022!):
```
data.dirFilterExcl = ['/store/unmerged/data/logs/prod/',
                                 '/store/unmerged/hidata/logs/prod/',
                                 '/store/unmerged/logs/prod/',
                                 '/store/unmerged/express/logs/prod/']
``` 
* once these new directories are in the service configuration, we can close this ticket.

What do you think?

**Comment by todor-ivanov** [comment]
Since we have not observed any problems with the current list of excluded directories, we are closing this issue and we are about to follow the last step in a separate one [1]

[1]
https://github.com/dmwm/WMCore/issues/10900
